### DISCUSSION TITLE
Github-copilot API

### DISCUSSION THREAD

**Discussion by ta32** [community/community#19240] [Copilot Conversations] [answered]
Hi there are extensions for Jetbrains and Visual studio IDE's. 

These extensions are obviously using an API to your service - are they documented somewhere so the community can develop their own extensions too?

Please document these internal API's 

The vscode extension js repo link is: 
"repository": {
  "type": "git",
  "url": "https://github.com/github/copilot-docs"
}
So I guess Microsoft doesn't want to open source its extensions so people can use the copilot API's?

**Comment by riversdiniz** [comment]
Guys, good morning!

When using and installing the copilot github extension is it necessary to do some configuration in vscode?

**Comment by hsenag** [comment, marked as answer]
This is something we are interested in doing at some point if we can, but I don't have a timeline at present. The APIs are still subject to quite a bit of change.

**Comment by qqdaiyu55** [comment]
+1 for API document proposal. @hsenag BTW does copilot support CORS request?

**Comment by MichaelReaNZ** [comment]
Access to API would be amazing, I'm already thinking how it could be integrated into a system.

**Comment by Nebula-Developer** [comment]
How is progress on this? I understand that copilot.vim is on GitHub, therefore to an extent API usage can be done currently but I'm struggling to find out how to port this to other languages.

**Comment by jeffwilliams** [comment]
I'd also like this. I want to integrate Copilot with an unsupported editor. Even a basic example of the JSONRPC calls needed would be helpful to get started.
